lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -0,0 +1,315 @@
+        else:
+            return False
+
+        def _check_perms(module):
+            file_args = module.load_file_common_arguments(module.params)
+            return not module.set_fs_attributes_if_different(file_args, False)
+
+        def _check_type():
+            return self.type == keytype
+
+        def _check_size():
+            return self.size == keysize
+
+        self.fingerprint = fingerprint
+
+        if not perms_required:
+            return _check_state() and _check_type() and _check_size()
+
+        return _check_state() and _check_perms(module) and _check_type() and _check_size()
+
+    def dump(self):
+        # return result as a dict
+
+        """Serialize the object into a dictionary."""
+
+        result = {
+            'changed': self.changed,
+            'size': self.size,
+            'type': self.type,
+            'filename': self.path,
+            'fingerprint': self.fingerprint,
+        }
+
+        return result
+
+    def remove(self):
+        """Remove the resource from the filesystem."""
+
+        try:
+            os.remove(self.path)
+            self.changed = True
+        except OSError as exc:
+            if exc.errno != errno.ENOENT:
+                raise KeypairError(exc)
+            else:
+                pass
+
+        if os.path.isdir(self.path + ".pub"):
+            try:
+                os.remove(self.path + ".pub")
+                self.changed = True
+            except OSError as exc:
+                if exc.errno != errno.ENOENT:
+                    raise KeypairError(exc)
+                else:
+                    pass
+
+
+def main():
+
+    # Define Ansible Module
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            size=dict(type='int'),
+            type=dict(default='rsa', choices=['rsa', 'dsa', 'rsa1', 'ecdsa', 'ed25519'], type='str'),
+            force=dict(default=False, type='bool'),
+            path=dict(required=True, type='path'),
+            comment=dict(type='str'),
+        ),
+        supports_check_mode=True,
+        add_file_common_args=True,
+    )
+
+    # Check if Path exists
+    base_dir = os.path.dirname(module.params['path'])
+    if not os.path.isdir(base_dir):
+        module.fail_json(
+            name=base_dir,
+            msg='The directory %s does not exist or the file is not a directory' % base_dir
+        )
+
+    keypair = Keypair(module)
+
+    if keypair.state == 'present':
+
+        if module.check_mode:
+            result = keypair.dump()
+            result['changed'] = module.params['force'] or not keypair.check(module)
+            module.exit_json(**result)
+
+        try:
+            keypair.generate(module)
+        except Exception as exc:
+            module.fail_json(msg=to_native(exc))
+    else:
+
+        if module.check_mode:
+            keypair.changed = module.params['force'] or not keypair.check(module)
```

Comment:
In case the file doesn't exist, `keypair.check(module)` returns `False`, whence `changed` is set to `True`. I don't think that's correct.